### PR TITLE
FLPATH-1078

### LINKED ISSUE #21: Question: Can the github workflow be more generic and not only used with Azure ?
## Question

The github flow which is created uses azure actions to set the k8s context and deploy the tekton resources. Such a workflow will only work on the Azure Cloud platform.

See: https://github.com/ch007m/onboarding/blob/main/.github/workflows/deploy.yaml#L18-L24

Could it be possible to only install such Tekton resources on azure if the user has selected Azure part of the template ?
Can such a flow be more generic and support to deploy the resources on an on-premises OCP cluster ?

### PR DESCRIPTION
**Pull request by dmartinol**
- closes #21 by using RH action instead of the Azure ones
- creates the ArgoCD application from the template
- creates the ArgoCD repository from the template

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -10,36 +10,30 @@ This repository is a collection of software and documentation templates for the
...
-
-**Note**: do not apply the manifests in the `kustomize/run` folder, as they refer to the same trigger resources that are automatically added by this software template.
+Finally, put the output of the last command in the `OPENSHIFT_TOKEN` secret.
+
+## Generated artifacts
+The execution of the software template produces the following output:
+* A workflow repository with a sample Serverless Workflow
+  * Including manifests to configure the Tekton trigger and ArgoCD application
+  * Includes a GH action to deploy the Tekton trigger and ArgoCD application: the action is automatically executed to start the deployment by the template
+  * A webHook is automatically created to activate the Tekton trigger
```
Comment (masayag) [inline README.md:35]: s/webHook/Github webhook ?

File: README.md [hunk 2]
```diff
@@ -10,36 +10,30 @@ This repository is a collection of software and documentation templates for the
...
+* A config repository with initial configuration of a kustomize project to deploy the application
+  * Uses properties file to allow setting user-specific configuration for the workflow (assuming that the workflow `application.properties` are using
+  env variables to specify the values)
+* Both the workflow and the config repos are defined as `Component` entities in Backstage
```
Comment (masayag) [inline README.md:39]: I'd rephrase this a bit to match backstage semantics to:
"Registers the workflow and workflow-config repositories in Backstage as `Components`.

File: scaffolder-templates/.github/workflows/deploy.yaml [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        uses: redhat-actions/oc-login@v1
+        with:
+          openshift_server_url: ${{ secrets.OPENSHIFT_SERVER }}
+          openshift_token: ${{ secrets.OPENSHIFT_TOKEN }}
+          namespace: default
+          insecure_skip_tls_verify: true
+
+      - name: Deploy Tekton trigger to the Kubernetes cluster
+        run: |
+          oc apply -n ${{ inputs.tektonNamespace }} -f tekton
```
Comment (masayag) [inline scaffolder-templates/.github/workflows/deploy.yaml:42]: Pls add / at the end to notate it is a folder, e.g. `-f tekton/`

File: scaffolder-templates/.github/workflows/deploy.yaml [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+      
+      - name: Deploy ArgoCD application to the Kubernetes cluster
+        run: |
+          oc apply -n ${{ inputs.argocdNamespace }} -f argocd
```
Comment (masayag) [inline scaffolder-templates/.github/workflows/deploy.yaml:46]: same here `argocd/`

File: scaffolder-templates/.github/workflows/deploy.yaml [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+      - name: Create ArgoCD repository
+        run: |
+          echo "${{ secrets.SSH_PRIVATE_KEY }}" > sshPrivateKey
+          cat sshPrivateKey
```
Comment (masayag) [inline scaffolder-templates/.github/workflows/deploy.yaml:50]: why this `cat` is needed?

Comment (dmartinol) [reply scaffolder-templates/.github/workflows/deploy.yaml:50]: Just to create a security breach 😬 
Removing it

**Review by masayag — APPROVED** [review]
